### ISSUE TITLE
Dictionaries cannot use longs as keys

### ISSUE THREAD

**Issue by The-Zenix** [sq/JSIL#725]
If you use a long as the key to a dictionary, attempting to access an element will always throw a keynotfoundexception.

eg.
            var dict = new Dictionary<long, string>();
            dict[4] = "hello";
            Logger.Log("value : " + dict[4]); //this line throws a keynotfoundexception

I'm guessing for this to work we would have to implement Int64.GetHashCode in javascript?

**Comment by kg** [comment]
Yeah, if they aren't hashable right now that's one likely candidate. It's also possible the Equals implementation is wrong.

**Comment by iskiselev** [comment]
#325 is the same issue.
